pkg/cli/build.go

```diff
@@ -39,12 +40,27 @@ func (s *Build) Run(cmd *cobra.Command, args []string) error {
 	if s.Push && (len(s.Tag) == 0 || s.Tag[0] == "") {
 		return fmt.Errorf("--push must be used with --tag")
 	}
+
 	c, err := s.client.CreateDefault()
 	if err != nil {
 		return err
 	}
 
+	if len(s.Tag) > 0 {
```

Comment:
If I remove that, the image will still be built, but it will throw an `invalid image tag` error while pushing. The expected behaviour mentioned in the issue is that `acorn build should detect the invalid --tag, show a friendly error on stderr, and avoid building the image`.

If you think it's okay to skip the part about not building the image, I can remove it. Let me know your thoughts.

Reply:
Sorry for the confusion, let me be more specific.

You are checking `if len(s.Tag) > 0 {` and then you have a `for _, tag := range s.Tag {`.

You don't need the if-statement because it is perfectly valid to have a for-loop over a `nil` or empty slice. If the slice is `nil` or empty, then the for-loop won't execute.

Reply:
Ah, I misunderstood. I thought you were referring to the entire code block, not just the if statement. I apologise for the confusion.

Reply:
Done.